### DISCUSSION TITLE
Enable Direct Download to BookPlayer

### DISCUSSION THREAD

**Discussion by bensaufley** [TortugaPower/BookPlayer#579] [Q&A] [answered]
I'm working on a Docker container to run on my NAS that will serve a little website that lets me download my audiobooks to any of my devices (and potentially more, once that's settled). This is helpful because i don't want to have to go to my computer every time, and because it gives me a better visual catalog and metadata sorting etc compared to just using the file system.

The POC works, except that trying to download directly to my iPhone doesn't seem to present a path to BookPlayer. I can download to my files, and `open with…` some things, but not BookPlayer. The file format (at least most commonly) is `.m4b`.

It's possible I'm missing something? Is there a way to do this? If not, is it something that might be possible? Honestly if you can point me in the right direction I'd be happy to take a crack at a PR myself.

**Comment by pichfl** [comment, marked as answer]
Check out our [URL Scheme](https://github.com/TortugaPower/BookPlayer/wiki/URL-scheme).

TLDR; Use `bookplayer://download?url="https://link-to/your/file.mp3"`

**Reply by bensaufley** [reply]
Oh that's awesome!! Thanks!
